qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -160,84 +180,203 @@ def __init__(
+            methods = []
+
+        for method in methods:
+            # There are two ways to specify a synthesis method. The more explicit
+            # way is to specify it as a tuple consisting of a synthesis algorithm and a
+            # list of additional arguments, e.g.,
+            #   ("kms", {"all_mats": 1, "max_paths": 100, "orig_circuit": 0}), or
+            #   ("pmh", {}).
+            # When the list of additional arguments is empty, one can also specify
+            # just the synthesis algorithm, e.g.,
+            #   "pmh".
+            if isinstance(method, tuple):
+                plugin_specifier, plugin_args = method
+            else:
+                plugin_specifier = method
+                plugin_args = {}
+
+            # There are two ways to specify a synthesis algorithm being run,
+            # either by name, e.g. "kms" (which then should be specified in entry_points),
+            # or directly as a class inherited from HighLevelSynthesisPlugin (which then
+            # does not need to be specified in entry_points).
+            if isinstance(plugin_specifier, str):
+                if plugin_specifier not in hls_plugin_manager.method_names(op.name):
+                    raise TranspilerError(
+                        "Specified method: %s not found in available plugins for %s"
+                        % (plugin_specifier, op.name)
+                    )
+                plugin_method = hls_plugin_manager.method(op.name, plugin_specifier)
+            else:
+                plugin_method = plugin_specifier
+
+            decomposition = plugin_method.run(
+                op,
+                coupling_map=self._coupling_map,
+                target=self._target,
+                qubits=qubits,
+                **plugin_args,
+            )
+
+            # The synthesis methods that are not suited for the given higher-level-object
+            # will return None, in which case the next method in the list will be used.
+            if decomposition is not None:
+                return decomposition
+
+        return None
+
+    def _synthesize_annotated_op(self, op: Operation) -> Union[Operation, None]:
+        """
+        Recursively synthesizes annotated operations.
+        Returns either the synthesized operation or None (which occurs when the operation
+        is not an annotated operation).
+        """
+        if isinstance(op, AnnotatedOperation):
+            # Currently, we ignore the qubits when recursively synthesizing the base operation.
+            synthesized_op, _ = self._recursively_handle_op(op.base_op, qubits=None)
+
+            # Currently, we depend on recursive synthesis producing either a QuantumCircuit or a Gate.
+            # If in the future we will want to allow HighLevelSynthesis to synthesize, say,
+            # a LinearFunction to a Clifford, we will need to rethink this.
+            if not synthesized_op or not isinstance(synthesized_op, (QuantumCircuit, Gate)):
```

Comment:
Could you use the bool second tuple element return from `_self.recursively_handle_op` here? But it doesn't really matter this should work fine.